### PR TITLE
Migrate krann and cf-Test from boost to catch2

### PR DESCRIPTION
**Pull request by jeffin143**
Refer #2523 

Let's throw 5 more test out of window, 47 to go :)
some_more

### CODE REVIEW

File: src/mlpack/tests/cf_test.cpp [hunk 1]
```diff
@@ -330,8 +328,9 @@ void Train(DecompositionPolicy& decomposition)
 
   for (size_t i = 0; i < combinations.n_cols; ++i)
   {
-    const double prediction = c.Predict(combinations(0, i), combinations(1, i));
```
Comment (birm) [inline src/mlpack/tests/cf_test.cpp:333]: This was exactly 80 characters, right?

File: src/mlpack/tests/cf_test.cpp [hunk 2]
```diff
@@ -450,68 +449,68 @@ void Serialization()
   SerializeObjectAll(c, cXml, cText, cBinary);
 
   // Check the internals.
-  BOOST_REQUIRE_EQUAL(c.NumUsersForSimilarity(), cXml.NumUsersForSimilarity());
-  BOOST_REQUIRE_EQUAL(c.NumUsersForSimilarity(),
+  REQUIRE(c.NumUsersForSimilarity() == cXml.NumUsersForSimilarity());
+  REQUIRE(c.NumUsersForSimilarity() ==
```
Comment (birm) [inline src/mlpack/tests/cf_test.cpp:453]: I think this also fits on one line

**Review by birm — COMMENTED** [review]
Waiting for the tests to resolve, but I had some line length notes

File: src/mlpack/tests/cf_test.cpp (hunk 1 above)
Comment (jeffin143) [reply src/mlpack/tests/cf_test.cpp:333]: no that was 81 chars

**Review by zoq — APPROVED** [review]
9UIpTz

File: src/mlpack/tests/cf_test.cpp (hunk 1 above)
Comment (birm) [reply src/mlpack/tests/cf_test.cpp:333]: :cry:

**Review by birm — APPROVED** [review]

**Pull request merged**